src/components/Header/BellNotification.jsx

```diff
@@ -0,0 +1,228 @@
+// Version: 1.2.3 - Added memoization, accessibility, error handling, and dark mode improvements
+import React, { useState, useEffect, useCallback, useRef, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+
+const NotificationBell = () => {
+  // State variables to manage notifications and UI state
+  const [hasNotification, setHasNotification] = useState(false);
+  const [isDataReady, setIsDataReady] = useState(false);
+  const [showNotification, setShowNotification] = useState(false);
+  const notificationRef = useRef(null);
+
+  // Fetching data from the Redux store
+  const timeEntries = useSelector((state) => state.timeEntries?.weeks?.[0] || []);
+  const weeklycommittedHours = useSelector((state) => state.userProfile?.weeklycommittedHours || 0);
+  const darkMode = useSelector((state) => state.theme.darkMode);
+  const userId = useSelector((state) => state.auth.user?.userid);
+
+  /**
+   * Memoized function to calculate the total effort (hours + minutes) logged by the user
+   */
+  const calculateTotalEffort = useMemo(() => {
+    return timeEntries.reduce((total, entry) => {
+      const hours = parseInt(entry.hours, 10) || 0;
+      const minutes = parseInt(entry.minutes, 10) || 0;
+      return total + hours + minutes / 60;
+    }, 0);
+  }, [timeEntries]);
+
+  /**
+   * Memoized function to calculate the time left until the end of the week (Sunday 7:00 AM UTC)
+   */
+  const calculateTimeLeft = useMemo(() => {
+    const currentTime = new Date();
+    const endOfWeek = new Date(currentTime);
```

Comment:
new Date(currentTime)  is returning same value as new Date(). what is the use of passing currentTime?

Reply:
Hey Sheetal, Good observation! You’re absolutely right; passing currentTime in that line is redundant since it's already a Date object. We could just use new Date() directly, which keeps the code a bit cleaner without changing any functionality. Thanks for catching that!